docs/machineregistration-reference.md

```diff
@@ -0,0 +1,126 @@
+|-------------|--------|--------------------------------------------------------------------------------------------------------------------------------------------|
+| firmware    | string | Firmware to install ('efi' or 'bios') (default "efi")                                                                                      |
+| device      | string | Device to install the system to                                                                                                            |
+| no-format   | bool   | Don’t format disks. It is implied that COS_STATE, COS_RECOVERY, COS_PERSISTENT, COS_OEM partitions are already existing on the target disk |
+| config-urls | list   | Cloud-init config files locations                                                                                                          |
+| iso         | string | Performs an installation from the ISO url instead of the running ISO                                                                       |
+| system-uri  | string | Sets the system image source and its type (e.g. 'docker:registry.org/image:tag') instead of using the running ISO                          |
+| debug       | bool   | Enable debug output                                                                                                                        |
+| tty         | string | Add named tty to grub                                                                                                                      |
+| poweroff    | bool   | Shutdown the system after install                                                                                                          |
+| reboot      | bool   | Reboot the system after install                                                                                                            |
+| eject-cd    | bool   | Try to eject the cd on reboot                                                                                                              |
+
+!!! warning
+    In case of using both `iso` and `system-uri` the `iso` value takes precedence
+
+The only required value for a successful installation is the `device` key as we need a target disk to install to. The rest of the parameters are all optional.
+
+??? example
+    ```yaml
+    apiVersion: elemental.cattle.io/v1beta1
+    kind: MachineRegistration
+    metadata:
+      name: my-nodes
+      namespace: fleet-default
+    spec:
+      config:
+        elemental:
+          install:
+            device: /dev/sda
+            debug: true
+            reboot: true
+            eject-cd: true
+            system-uri: registry.opensuse.org/isv/rancher/elemental/teal52/15.3/rancher/elemental-node-image/5.2:latest
+    ```
+
+
+#### machineName
+
+This refers to the name that will be set to the node and the kubernetes resources that require a hostname (rk2 pods for example)
```

Comment:
> kubernetes resources that require a hostname (rk2 pods for example)

I am a bit lost here... What do you mean with this?   Which rk2 pods? 🤔

Reply:
When you deploy with rke2, some of the pods have the hostname of the node appended to the pod name